app/actions/handlers.py

```diff
@@ -246,32 +246,35 @@ async def get_integrated_alerts(integration:Integration, action_config: PullEven
               for geostore_id in geostore_ids 
               for lower, upper in generate_date_pairs(start_date, end_date)]
 
-    for t in asyncio.as_completed(tasks):
-        integrated_alerts = await t
-        if integrated_alerts:
-            logger.info(f"Integrated alerts pulled with success.")
-            transformed_data = [transform_integrated_alert(alert)for alert in integrated_alerts]
-            await handle_transformed_data(
-                transformed_data,
-                str(integration.id),
-                "pull_events"
-            )
-            total_alerts += len(integrated_alerts)
-
-    dataset_status = DatasetStatus(
-        dataset=dataset_metadata.dataset,
-        version=dataset_metadata.version,
-        latest_updated_on=dataset_metadata.updated_on
-    )
+    try:
+        for t in asyncio.as_completed(tasks):
+            integrated_alerts = await t
+            if integrated_alerts:
+                logger.info(f"Integrated alerts pulled with success.")
+                transformed_data = [transform_integrated_alert(alert)for alert in integrated_alerts]
+                await handle_transformed_data(
+                    transformed_data,
+                    str(integration.id),
+                    "pull_events"
+                )
+                total_alerts += len(integrated_alerts)
+    except QuotaExceeded:
+        logger.warning(f"Quota exceeded for {auth_config.email} on dataset: {DATASET_GFW_INTEGRATED_ALERTS}.")
```

Comment:
We could generate an activity log with `error` level, or re-raise the exception here if we want this to flag the related connection as unhealthy, and then we would also see something in activity logs.